src/api/improve.rkt

```diff
@@ -33,7 +33,15 @@
 (define (run-improve input output #:threads [threads #f])
   (define seed (get-seed))
   (define tests (load-tests input))
-  (define results (get-test-results tests #:threads threads #:seed seed #:profile #f #:dir #f))
+  (start-job-server threads)
+  (define-values (ids)
+    (for/list ([test tests])
```

Comment:
In general please do `(in-list tests)` instead of just `tests`; it's faster and also acts like a type hint.